pinot-common/src/main/java/org/apache/pinot/common/utils/DataTable.java

```diff
@@ -56,6 +57,8 @@
 
   double getDouble(int rowId, int colId);
 
+  BigDecimal getBigDecimal(int rowId, int colId);
```

Comment:
For data access layer, we should not introduce the `BigDecimal` type, but use the stored type (see how `Timestamp` is handled)